Lib/test/test_types.py

```diff
@@ -694,6 +695,15 @@ def test_or_type_operator_with_NewType(self):
     def test_or_type_operator_with_IO(self):
         assert typing.IO | str == typing.Union[typing.IO, str]
 
+    def test_or_type_operator_with_SpecialForm(self):
+        assert typing.Any | str == typing.Union[typing.Any, str]
+        assert typing.NoReturn | str == typing.Union[typing.NoReturn, str]
+        assert typing.Optional[int] | str == typing.Union[typing.Optional[int], str]
+        assert typing.Union[int, bool] | str == typing.Union[typing.Union[int, bool], str]
+
+    def test_or_type_repr(self):
+        assert repr(int | None) == "int | None"
```

Comment:
Hm, I think this should be fixed at the `__args__` level -- `(int | None).__args__ == (int, None)`. At least that would be more consistent with PEP 585: `list[None].__args__ == (None,)`.

(OTOH I just discovered that `typing.Optional[int]` does put `NoneType` in `__args__`, so now I'm not sure. This is probably not specified anywhere.)

Reply:
It looks like replacing `None` with `type(None)` is done explicitly in many cases. Example: https://github.com/python/cpython/blob/master/Lib/typing.py#L138

Do you happen to know why this was done?

Reply:
Unfortunately I don't recall why. It has been there for a very long time. We didn't follow this for the new `GenericAlias` type though. If you just keep the `None` everywhere for the new `Union` type, does anything fail? I guess you would have to special-case it for `isinstance()` and `issubclass()` -- how ugly would that look?

Another approach might be to change `GenericAlias` to do the replacement of `None` with `type(None)`. (Then again, the new `GenericAlias` is intentionally pretty dumb -- you can write `list[42]` without getting an error. But for the new `Union` we'll have to have somewhat higher standards, due to the desire for `isinstance()` and `issubclass()` support (we intentionally disallow `GenericAlias` there).

Reply:
Okay, so I've sorted out where I was going wrong here. 

In order to make sure that equality and isinstance work as we'd previously defined, I had to add some extra handling for the `None` type. Let me know if that's not going to work long term.